### ISSUE TITLE
Need Option to not override window.postMessage

### ISSUE THREAD

**Issue by schlaegerz** [CRAlpha/react-native-wkwebview#170]
I have a web app that I am trying to render in this Webview that uses Meteor, which uses window.postMessage whenever I set onMessage the Webapp stops working as expected. 

What would be great would be some option to just not override postMessage and use the namespaced version.

The Webview in the core react native has a similar issue.

**Comment by insraq** [comment]
I believe it will only override postMessage if you have `onMessage` callback. It's quite easy to change it though but it won't be "compatible" with RN's WebView

**Comment by schlaegerz** [comment]
Yeah, i guess that makes sense, I'll figure out something on my own, I've been trying to push the core react-native to take a similar change and doing this would mean i would need to figure out something for android anyways.

**Comment by shirakaba** [comment]
I lost a whole day testing traditional usage of `window.postMessage()` to post messages to iframes, and found that it was impossible if the `onMessage` prop was also in use.

We can have both working simultaneously.

Currently, the following code is injected upon the `.atDocumentEnd` hook (yes, it's horrible, but it's consistent with the [horrible behaviour of RN's `<WebView>` component](https://github.com/facebook/react-native/blob/26684cf3adf4094eb6c405d345a75bf8c7c0bf88/React/Views/RCTWebView.m#L313)):

```js
window.originalPostMessage = window.postMessage;
window.postMessage = function() {
    return window.webkit.messageHandlers.reactNative.postMessage.apply(window.webkit.messageHandlers.reactNative, arguments);
};
```

As you can see, we retain a reference to `window.originalPostMessage`.

The `window.postMessage()` [docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) show that it is invoked with two to three parameters:

```
targetWindow.postMessage(message, targetOrigin, [transfer]);
```

RN's `window.postMessage()`, by contrast, specifically accepts only the first param, and incidentally that must be a string (rather than any type that works with the [Structured Clone Algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)).

So we can both maintain consistency with the core *and* un-break `window.postMessage()` for traditional usages:

```js
window.originalPostMessage = window.postMessage;
window.postMessage = function(message, targetOrigin, transfer) {
	return typeof targetOrigin === 'undefined' ? 
		window.webkit.messageHandlers.reactNative.postMessage(message) : 
		window.originalPostMessage(message, targetOrigin, transfer);
};
```

What do you think?

**Comment by shirakaba** [comment]
Okay, I've created a PR #176 for this :)
